app/(root)/post/create/page.tsx

```
@@ -0,0 +1,22 @@
+import { auth } from '@/auth';
+import CreateOrUpdatePost from '@/components/post/CreateOrUpdatePost';
+import { getTags } from '@/lib/actions/tag-actions';
+import type { ITag } from '@/types/tag';
+
+// ----------------------------------------------------------------
+
+const CreatePostPage = async () => {
+  const session = await auth();
+  if (!session?.user.id) return;
+
+  const tags: ITag[] = (await getTags(session.user.id)) || [];
```

It would be better to throw an error if the tags return undefined, since they are a crucial piece of the information, instead of defaulting to an empty array with no tags.